components/SegmentedControl/src/EqualPanel.cs

```diff
@@ -37,18 +37,26 @@ protected override Size MeasureOverride(Size availableSize)
 
         if (Children.Count > 0)
         {
-            SetMaxDimensions(Children, availableSize);
+            foreach (var child in Children)
+            {
+                child.Measure(availableSize);
+                maxItemWidth = Math.Max(maxItemWidth, child.DesiredSize.Width);
+                maxItemHeight = Math.Max(maxItemHeight, child.DesiredSize.Height);
+            }
 
-            // Equal columns based on the available width
-            if (HorizontalAlignment == HorizontalAlignment.Stretch)
+            // Return equal widths based on the widest item
+            // In very specific edge cases the AvailableWidth might be infinite resulting in a crash.
+            if (HorizontalAlignment != HorizontalAlignment.Stretch || double.IsInfinity(availableSize.Width))
```

Comment:
Future: Thinking long-term, maybe we should create an issue to track adding an `Orientation` property here so it can do equal heights as well? Then this would be VerticalAlignment and height as a flip on Orientation here as well?

I'm trying to remember why `UniformGrid` didn't work again, but that's a lot heavier-weight anyway... so simple panel is simple. 😋